### PR TITLE
refactor(frontend): eliminate `PlanNode` rest panic

### PR DESCRIPTION
**Pull request by likzn**
## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

- Summarize your change (**mandatory**)
This is a follow PR to eliminate rest `panic` where the previous forget.
Issue: #2374 
Previous PR: #2396 

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

### CODE REVIEW

**Comment by likzn** [comment]
@fuyufjh PTAL. You can check the d82ff3c414ef763a1fdf570c4a682a68a49e4188, it is the real change.

File: src/frontend/src/optimizer/plan_node/logical_delete.rs [hunk 1]
```diff
@@ -101,10 +101,14 @@ impl ToBatch for LogicalDelete {
 
 impl ToStream for LogicalDelete {
     fn to_stream(&self) -> Result<PlanRef> {
-        unreachable!("delete should always be converted to batch plan");
+        Err(RwError::from(ErrorCode::InternalError(
+            "delete should always be converted to batch plan".to_string(),
+        )))
```
Comment (fuyufjh) [inline src/frontend/src/optimizer/plan_node/logical_delete.rs:106]: I think this should be kept as it is, because it works like an assertion

Comment (xxchan) [reply src/frontend/src/optimizer/plan_node/logical_delete.rs]: > If the latter,the panic is the strong contrains may be better.

What do you mean by "strong contrains"? `unreachable` is just a special kind of `panic`

Comment (likzn) [reply src/frontend/src/optimizer/plan_node/logical_delete.rs]: OK.If i am product user. And i run a sql and it panic. Is there some error to tell me what happened and what should i do next?

Comment (likzn) [reply src/frontend/src/optimizer/plan_node/logical_delete.rs]: Get

File: src/frontend/src/optimizer/plan_node/logical_join.rs [hunk 2]
```diff
@@ -423,7 +422,10 @@ impl ToStream for LogicalJoin {
             }
         } else {
             // Convert to Nested-loop Join for non-equal joins
-            todo!("nested loop join")
+            Err(RwError::from(ErrorCode::NotImplemented(
+                "nested loop join".to_string(),
+                None.into(),
```
Comment (BugenZhao) [inline src/frontend/src/optimizer/plan_node/logical_join.rs:427]: Do we have a tracking issue for this? It's better to fill the issue ID here. cc @fuyufjh

**Comment by fuyufjh** [comment]
@likzn Please solve the conflicts

**Comment by likzn** [comment]
@fuyufjh Done

File: src/frontend/src/optimizer/plan_node/logical_join.rs (hunk 2 above)
Comment (fuyufjh) [reply src/frontend/src/optimizer/plan_node/logical_join.rs]: Created a issue just now #2655

**Pull request merged**